src/util.c

```diff
@@ -58,12 +64,22 @@ init_log(void)
 	}
 }
 
+static void
+append_log(const char *s, size_t len)
+{
+	assert(logbuffer);
+	if (logsize + len > logsize) {
+		logbuffer = x_realloc(logbuffer, logsize + len + 1);
+		logsize = logsize + len;
+	}
+	strncat(logbuffer, s, len);
```

Comment:
O(n²) here as well – it would be nice to update and use a buffer size variable instead of implicitly `strlen`-ing the potentially large string for each call.

Reply:
Hmm, I don't think `strcat` calls `strlen` does it ? Was careful to use the `len` parameter, actually.

Reply:
It doesn't call it on the *source* string, but it does call it on the *destination* string. My bad.